File: leetcode/1001-2000/1801-1900/1871-1880/1876-substrings-of-size-three-with-distinct-characters/SubstringsOfSizeThreeWithDistinctCharacters.java
public class SubstringsOfSizeThreeWithDistinctCharacters {

  public static void main(String[] args) {
    var solution = new Solution();

    System.out.println(solution.countGoodSubstrings("aababcabc"));
    System.out.println(solution.countGoodSubstrings("xyzzaz"));
  }
}

class Solution {
  public int countGoodSubstrings(String s) {
    if (s.length() < 3) {
      return 0;
    }

    var totalGoodSubstrings = 0;

    var b = s.charAt(0);
    var c = s.charAt(1);
    var a = ' ';

    for (var i = 2; i < s.length(); i++) {
      a = b;
      b = c;
      c = s.charAt(i);

      if (a != b && b != c && c != a) {
        totalGoodSubstrings++;
      }
    }

    return totalGoodSubstrings;
  }
}

File: leetcode/1001-2000/1801-1900/1871-1880/1876-substrings-of-size-three-with-distinct-characters/SubstringsOfSizeThreeWithDistinctCharacters.ts
function countGoodSubstrings(s: string): number {
  let totalGoodSubstrings = 0;

  for (let i = 0; i < s.length - 2; i++) {
    const a = s[i];
    const b = s[i + 1];
    const c = s[i + 2];

    if (a !== b && b !== c && c !== a) {
      totalGoodSubstrings++;
    }
  }

  return totalGoodSubstrings;
};
